cancer_ai/validator/model_manager.py

```diff
@@ -262,7 +270,66 @@ def get_pioneer_models(self, grouped_hotkeys: list[list[str]]) -> list[str]:
                     continue
 
                 try:
-                    block_timestamp = self.db_controller.get_block_timestamp(model_info.block)
+                    matches = fs.glob(f"{model_info.hf_repo_id}/extrinsic_record.json", refresh=True)
+                    if not matches:
+                        raise FileNotFoundError("extrinsic_record.json not found in repo")
+
+                    record_path = matches[0]
+                    with fs.open(record_path, "r", encoding="utf-8") as rf:
+                        record_data = json.load(rf)
+
+                    file_hotkey = record_data.get("hotkey")
+                    extrinsic_id = record_data.get("extrinsic")
+                    if file_hotkey != hotkey or not extrinsic_id:
+                        raise ValueError(f"Invalid record contents: {record_data}")
+
+                except Exception as e:
+                    bt.logging.error(f"Failed to load HF repo extrinsic record for {hotkey}: {e}")
+                    self.parent.error_results.append((hotkey, "Invalid or missing extrinsic record in HF repo."))
+                    continue
+
+                try:
+                    blk_str, idx_str = extrinsic_id.split("-", 1)
```

Comment:
mało czytelne, wygląda jak magic code
funkcja jest za długa, podziel to jakoś na funkcje czy wydziel  / zrób komentarze

Reply:
przecież to zwykły python, splituje string na dwie zmienne dashem... podstawy jakiegoś basic leetcode'a ;p 
funkcja jest długa to prawda